.github/workflows/ci.yml

```diff
@@ -46,18 +43,17 @@ jobs:
     - name: Install Mage
       run: go install github.com/magefile/mage
 
-    - name: Update to latest spec
-      shell: bash
-      run: |
-        export PATH=$PATH:$(go env GOPATH)/bin
-        go install github.com/swaggo/swag/cmd/swag@v1.8.7
-        mage spec
-
-    - name: Push generated spec file
+    - name: Update spec and push generated spec file
       uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.head.ref }}
```

Comment:
this workflow runs not just on PRs but on merges to main - so this would fail at that point as there is no pull_request.head. 

I think this should be pulled out into its own workflow for swagger publish.